### PR TITLE
Added a new (runtime only) property "dontuseqt" to the player obj

### PR DESCRIPTION
**Pull request by livecodepanos**

### CODE REVIEW

**Comment by livecodepanos** [comment]
Not sure why the commit has Michael's name. I guess because I did this commit from my Windows VM, which has only Michael's account :)

**Comment by livecodepanos** [comment]
This pull request comes together with https://github.com/runrev/livecode-ide/pull/235

File: engine/src/objdefs.h [hunk 1]
```diff
@@ -214,6 +214,7 @@ along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
 #define F_PLAY_SELECTION        (1UL << 20)
 #define F_SHOW_SELECTION        (1UL << 21)
 #define F_SHOW_VOLUME           (1UL << 22)
+#define F_DONTUSEQT             (1UL << 23)
```
Comment (runrevmark) [inline engine/src/objdefs.h]: As this is a runtime only property - there's no need to allocate a F_ flag for it.

File: engine/src/player-legacy.cpp [hunk 2]
```diff
@@ -1018,6 +1023,16 @@ Exec_stat MCPlayer::setprop(uint4 parid, Properties p, MCExecPoint &ep, Boolean
 		if (dirty)
 			setlooping((flags & F_LOOPING) != 0); //set/unset movie looping
 		break;
+
+	case P_DONT_USE_QT:
+		if (!MCU_matchflags(data, flags, F_DONTUSEQT, dirty))
```
Comment (runrevmark) [inline engine/src/player-legacy.cpp]: I'm not sure I understand the use of the flags here - we have a member variable with the value in it...

**Comment by runrevmark** [comment]
Having pondered this then - dontuseqt should be false by default. When the Player's dontuseqt is false, then it uses MCdontuseqt. If dontuseqt is true then it won't use QT.

Now, I think we've got something slightly wrong here - the point of choice as to whether to use QT or not is at the point the movie is prepared. After that point, if it used QT then all the QT code paths have to be followed; if it did not use QT then the other codepaths need to be followed.

Thus I suggest we need 'dontuseqt' as a member variable _and_ usingqt as member variable.

At the point of prepare:

```
If dontuseqt is false and MCdontuseqt is false and qt is present then
   prepare qt movie
   usingqt = true
else
  prepare other type of movie
  usingqt = false
end if
```

At all other points which do stuff with a prepared movie, use usingqt to determine what to do.

This makes the granularity of the decision to use qt or not for a given player centered around when a movie is re-prepared.

File: engine/src/player-legacy.cpp [hunk 3]
```diff
@@ -218,6 +218,8 @@ MCPlayer::MCPlayer()
 	userCallbackStr = NULL;
 	formattedwidth = formattedheight = 0;
 	loudness = 100;
+	dontuseqt = MCdontuseQT;
```
Comment (runrevmark) [inline engine/src/player-legacy.cpp]: This isn't quite what I suggested.

dontuseqt is initialized to false - if it is false at the point a movie is prepared _then_ the value of MCdontuseqt is used

File: engine/src/player-legacy.cpp [hunk 4]
```diff
@@ -218,6 +218,8 @@ MCPlayer::MCPlayer()
...
+	usingqt = !MCdontuseQT;
```
Comment (runrevmark) [inline engine/src/player-legacy.cpp]: this should be initialized to false by default - it doesn't mean anything until the movie is prepared

File: engine/src/player-legacy.cpp [hunk 5]
```diff
@@ -387,7 +391,7 @@ Boolean MCPlayer::kdown(const char *string, KeySym key)
 
 #ifndef FEATURE_PLATFORM_PLAYER
 #ifdef FEATURE_QUICKTIME
-	if (qtstate == QT_INITTED && state & CS_PREPARED)
+	if (mustUseQT() && state & CS_PREPARED)
```
Comment (runrevmark) [inline engine/src/player-legacy.cpp]: I'd suggest renaming 'mustUseQT()' to 'usingQT()'

**Comment by runrevmark** [comment]
This looks good - it just needs a release note and an update to the dontUseQT property dictionary entry. This is quite possibly a feature people are going to want to use :)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 80a0e66

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 80a0e66a8458d6a51e1bcc4622a494b76efc3958

**Pull request merged**